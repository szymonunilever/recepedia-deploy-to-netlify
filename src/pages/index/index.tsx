import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
