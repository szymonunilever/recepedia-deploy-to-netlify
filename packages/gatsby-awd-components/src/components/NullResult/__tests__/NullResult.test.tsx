import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { NullResult } from '../index';
import toJson from 'enzyme-to-json';
const contentVariants = [
  {
    title: 'Oops! No results',
    subtitle: 'Maybe try the following:',
    textList: [`Don't use too many filters at once`, `Try using only filters`],
  },
  {
    subtitle: 'Maybe try the following:',
    textList: [`Don't use too many filters at once`, `Try using only filters`],
  },
  {
    title: 'Oops! No results',
    textList: [`Don't use too many filters at once`, `Try using only filters`],
  },
];
describe('<NullResult />', () => {
  let wrapper: ReactWrapper;
  it('matches the snapshot', () => {
    contentVariants.forEach(content => {
      wrapper = mount(<NullResult {...{ content }} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
