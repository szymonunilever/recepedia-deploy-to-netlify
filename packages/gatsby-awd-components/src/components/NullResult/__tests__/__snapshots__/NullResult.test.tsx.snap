// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NullResult /> matches the snapshot 1`] = `
<Component
  content={
    Object {
      "subtitle": "Maybe try the following:",
      "textList": Array [
        "Don't use too many filters at once",
        "Try using only filters",
      ],
      "title": "Oops! No results",
    }
  }
>
  <div
    className="wrapper recommendations"
    data-componentname="nullResult"
    data-componentview="nullResult"
  >
    <Text
      className="recommendations__title recommendations__title"
      tag={1}
      text="Oops! No results"
    >
      <h2
        className="recommendations__title recommendations__title"
      >
        Oops! No results
      </h2>
    </Text>
    <Text
      className="recommendations__subtitle recommendations__subtitle"
      tag={2}
      text="Maybe try the following:"
    >
      <h3
        className="recommendations__subtitle recommendations__subtitle"
      >
        Maybe try the following:
      </h3>
    </Text>
    <ul
      className="recommendations__list recommendations__list"
    >
      <li
        className="recommendations__item recommendations__item"
        key="0"
      >
        Don't use too many filters at once
      </li>
      <li
        className="recommendations__item recommendations__item"
        key="1"
      >
        Try using only filters
      </li>
    </ul>
  </div>
</Component>
`;

exports[`<NullResult /> matches the snapshot 2`] = `
<Component
  content={
    Object {
      "subtitle": "Maybe try the following:",
      "textList": Array [
        "Don't use too many filters at once",
        "Try using only filters",
      ],
    }
  }
>
  <div
    className="wrapper recommendations"
    data-componentname="nullResult"
    data-componentview="nullResult"
  >
    <Text
      className="recommendations__subtitle recommendations__subtitle"
      tag={2}
      text="Maybe try the following:"
    >
      <h3
        className="recommendations__subtitle recommendations__subtitle"
      >
        Maybe try the following:
      </h3>
    </Text>
    <ul
      className="recommendations__list recommendations__list"
    >
      <li
        className="recommendations__item recommendations__item"
        key="0"
      >
        Don't use too many filters at once
      </li>
      <li
        className="recommendations__item recommendations__item"
        key="1"
      >
        Try using only filters
      </li>
    </ul>
  </div>
</Component>
`;

exports[`<NullResult /> matches the snapshot 3`] = `
<Component
  content={
    Object {
      "textList": Array [
        "Don't use too many filters at once",
        "Try using only filters",
      ],
      "title": "Oops! No results",
    }
  }
>
  <div
    className="wrapper recommendations"
    data-componentname="nullResult"
    data-componentview="nullResult"
  >
    <Text
      className="recommendations__title recommendations__title"
      tag={1}
      text="Oops! No results"
    >
      <h2
        className="recommendations__title recommendations__title"
      >
        Oops! No results
      </h2>
    </Text>
    <ul
      className="recommendations__list recommendations__list"
    >
      <li
        className="recommendations__item recommendations__item"
        key="0"
      >
        Don't use too many filters at once
      </li>
      <li
        className="recommendations__item recommendations__item"
        key="1"
      >
        Try using only filters
      </li>
    </ul>
  </div>
</Component>
`;
